File: src/main/java/com/crud/company/service/DepartmentService.java
package com.crud.company.service;

import com.crud.company.entity.Department;

import java.util.List;

public interface DepartmentService {
    List<Department> getAllDepartments();
    Department addDepartment(Department department);
    Department updateDepartment(Department department);
    void deleteDepartment(Long id);
}

File: web/src/app/department/department.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class DepartmentService {

  constructor(private http:HttpClient) { }
  baseUrl:string = "http://localhost:8080/department";

  getDepartments(){
    return this.http.get(this.baseUrl);
  }
  addDepartment(department:any){
    return this.http.post(this.baseUrl, department);
  }
  updateDepartment(department:any){
    return this.http.put(this.baseUrl, department);
  }
  deleteDepartment(id:number){
    return this.http.delete(this.baseUrl+"/"+id);
  }
}
